File: 17.js
class StringHashTable {
    constructor(size = 10) {
        this.size = size;
        this.table = Array.from({ length: size }, () => []);
    }

    hash(str) {
        let sum = 0;
        for (let c of str) sum += c.charCodeAt(0);
        return sum % this.size;
    }

    insert(key) {
        const idx = this.hash(key);
        if (!this.table[idx].includes(key)) {
            this.table[idx].push(key);
        }
    }

    search(key) {
        return this.table[this.hash(key)].includes(key);
    }

    getChain(key) {
        return this.table[this.hash(key)];
    }
}

File: 17.py
class StringHashTable:
    def __init__(self, size=10):
        self.size = size
        self.table = [[] for _ in range(size)]

    def hash(self, s):
        return sum(ord(c) for c in s) % self.size

    def insert(self, key):
        idx = self.hash(key)
        if key not in self.table[idx]:
            self.table[idx].append(key)

    def search(self, key):
        return key in self.table[self.hash(key)]

    def get_chain(self, key):
        return self.table[self.hash(key)]
